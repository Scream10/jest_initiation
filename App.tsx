import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function App() {
  type User = {
    createdAt: Date;
    id: number;
    name: string;
  }

  const user: User = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  return (
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
      <Text>My name is {user.name}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
