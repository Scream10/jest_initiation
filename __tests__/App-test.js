import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

test('render correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('check and compare user', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
    name: expect.any(String)
  });
});
  