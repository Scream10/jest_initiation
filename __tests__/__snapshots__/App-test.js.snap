// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check and compare user 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": Any<String>,
}
`;

exports[`render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.js to start working on your app!
  </Text>
  <Text>
    My name is 
    LeBron James
  </Text>
</View>
`;
